src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemDataAutomationPeer.cs

```diff
@@ -0,0 +1,166 @@
+                return wrapperPeer.GetAutomationControlType();
+            }
+
+            return AutomationControlType.MenuItem;
+        }
+
+        public override object GetPattern(PatternInterface patternInterface)
+        {
+            switch (patternInterface)
+            {
+                case PatternInterface.ExpandCollapse:
+                case PatternInterface.Invoke:
+                case PatternInterface.Toggle:
+                    return this;
+            }        
+
+            return base.GetPattern(patternInterface);
+        }
+
+        private void EnsureEnabled()
+        {
+
+            FrameworkElementAutomationPeer itemsControllerPeer = GetItemsControlAutomationPeer();
+            if (!itemsControllerPeer.IsEnabled())
+            {
+                throw new ElementNotEnabledException();
+            }
+        }
+
+        #endregion
+
+        #region IExpandCollapseProvider Members
+
+        void IExpandCollapseProvider.Expand()
+        {
+            MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+            if (wrapperPeer != null)
+            {
+                IExpandCollapseProvider expandCollapseProvider = wrapperPeer as IExpandCollapseProvider;
+                expandCollapseProvider.Expand();
+            }
+            ThrowElementNotAvailableException();
+        }
+
+        void IExpandCollapseProvider.Collapse()
+        {
+            MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+            if (wrapperPeer != null)
+            {
+                IExpandCollapseProvider expandCollapseProvider = wrapperPeer as IExpandCollapseProvider;
+                expandCollapseProvider.Collapse();
+            }
+            ThrowElementNotAvailableException();
+        }
+
+        ExpandCollapseState IExpandCollapseProvider.ExpandCollapseState
+        {
+            get
+            {
+                MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+                if (wrapperPeer != null)
+                {
+                    IExpandCollapseProvider expandCollapseProvider = wrapperPeer as IExpandCollapseProvider;
+                    return expandCollapseProvider.ExpandCollapseState;
+                }
+                ThrowElementNotAvailableException();
+                return ExpandCollapseState.LeafNode;
+            }
+        }
+
+        #endregion
+
+        #region IInvokeProvider Members
+
+        void IInvokeProvider.Invoke()
+        {
+            // check if enabled
+            EnsureEnabled();
+
+            MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+            if (wrapperPeer != null)
+            {
+                IInvokeProvider invokeProvider = wrapperPeer as IInvokeProvider;
+                invokeProvider.Invoke();
+            }
+            else
+            {
+                ThrowElementNotAvailableException();
+            }
```

Comment:
```suggestion
            if (wrapperPeer is invokeProvider )
            {
                invokeProvider.Invoke();
            }
            else
            {
                ThrowElementNotAvailableException();
            }
```

The origin code may throw NullReferenceException when the wrapperPeer is not the IInvokeProvider(actually this won't happen).

Reply:
I'll change this to an explicit cast. This may is faster and more intentional.